compiler-rt/test/builtins/Unit/fp_test.h

```diff
@@ -230,7 +230,7 @@ static inline double makeQNaN64(void)
     return fromRep64(0x7ff8000000000000UL);
 }
 
-#if __LDBL_MANT_DIG__ == 64 && defined(__x86_64__)
+#if HAS_80_BIT_LONG_DOUBLE
```

Comment:
Previously, we specifically checked for `x86_64` here, but `HAS_80_BIT_LONG_DOUBLE` also is true on i686 mingw, so this will start executing these codepaths on i686 now as well.

Reply:
Yeah, https://github.com/mstorsjo/llvm-project/actions/runs/12026429539/job/33525399943 this suggests uint128t is not defined, thus this should be only enabled for x86_64.
Alternatively we could check if __uint128t is available with check_c_source_compiles and we can add another flag like we do here https://github.com/llvm/llvm-project/blob/f082782c1b3ec98f50237ddfc92e6776013bf62f/compiler-rt/lib/builtins/CMakeLists.txt#L873.

Reply:
FWIW, I also tested with an addition like this: https://github.com/mstorsjo/llvm-project/commit/de5ad3a19cf3c4d23bf1e338d27447d61f2eb731

That unbreaks the other unrelated tests, but this new test still is failing: https://github.com/mstorsjo/llvm-project/actions/runs/12026756315/job/33526363616

```
# | D:\a\llvm-project\llvm-project\compiler-rt\test\builtins\Unit\extendhfxf2_test.c:16:13: error: call to undeclared function 'compareResultF80'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
# |    16 |   int ret = compareResultF80(x, expectedHi, expectedLo);
# |       |             ^
```
So a similar condition would need to be added back to the condition within the actual test here as well.